arch/arc/core/thread.c

```diff
@@ -253,3 +253,37 @@ int arch_float_enable(struct k_thread *thread, unsigned int options)
 	return 0;
 }
 #endif /* CONFIG_FPU && CONFIG_FPU_SHARING */
+
+#if defined(CONFIG_DSP) && defined(CONFIG_DSP_SHARING)
+int arch_dsp_disable(struct k_thread *thread)
+{
+	unsigned int key;
+
+	/* Ensure a preemptive context switch does not occur */
+
+	key = irq_lock();
```

Comment:
Is it OK to use irq_lock and not a spinlock here? Are we expect that this code can be used on SMP platform in the manner that we are calling this function for thread != _current?

Reply:
fixed